merge-two-sorted-lists/std-freejia.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * 고생한 이유: ListNode 의 시작점을 따로 둬야 하는 것!
+ * 
+ * ListNode dummy = new ListNode(0); // 시작점
+ * ListNode output = dummy; // 시작점 뒤에 실제 작업용 포인터
+ * 
+ * 
+ * Definition for singly-linked list.
+ * public class ListNode {
+ *     int val;
+ *     ListNode next;
+ *     ListNode() {}
+ *     ListNode(int val) { this.val = val; }
+ *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
+ * }
+ */
+class Solution {
+    public static ListNode mergeTwoLists(ListNode list1, ListNode list2) {
+        // 길이가 둘다 null 이면, return null
+        if (list1 == null && list2 == null) return null;
+
+        ListNode dummy = new ListNode(0); // 시작점
+        ListNode output = dummy; // 시작점 뒤에 실제 작업용 포인터
```

Comment:
```suggestion
        ListNode dummy = new ListNode(); // 시작점
```

둘다 똑같은 시작점이므로 `// 시작점 뒤에 실제 작업용 포인터`라는 설명은 틀린 것 같아요.
한 줄로 합쳐도 되겠는데 명시해 주신 것 같네요.
또한 0을 안 넘겨도 무방하겠군요.

Reply:
dummy 는 고정된 시작점이고, output 은 이어붙일 포인터라서 따로 관리했습니다. 
한 줄로 합치려고 했는데, accept 되지 못했습니다

Reply:
오, 맞네요! 제가 잘못 생각했습니다.
`output`은 Iterator의 역할이군요.